File: src/client/src/types/attributes/valueConstraintView.ts
import { HasCardinality } from "types/common/hasCardinality";
import { ConstraintType } from "./constraintType";
import { XsdDataType } from "./xsdDataType";

export interface ValueConstraintView extends HasCardinality {
  constraintType: ConstraintType;
  dataType: XsdDataType;
  value: string | number | boolean | null;
  valueList: string[] | number[] | null;
  pattern: string | null;
  minValue: number | null;
  maxValue: number | null;
}

File: src/server/Tyle.Api/Attributes/ValueConstraintView.cs
using Tyle.Core.Attributes;

namespace Tyle.Api.Attributes;

public class ValueConstraintView
{
    public ConstraintType ConstraintType { get; set; }
    public XsdDataType DataType { get; set; }
    public int? MinCount { get; set; }
    public int? MaxCount { get; set; }
    public string? Pattern { get; set; }
    public decimal? MinValue { get; set; }
    public decimal? MaxValue { get; set; }
}